### PR TITLE
bugfix - Button Border

### PR DESCRIPTION
**Pull request by aletail**
### Testing Steps
- [ ] Confirm you can set the border of a button component:
```
$component-button-properties: (
  ...
  border: solid 1px map-get($theme-properties, primaryColor),
  ...
);
```

### Issues/Concerns
- I think I have the labels right for this, but please let me know other wise

### CODE REVIEW

**Review by tiller1010 — APPROVED** [review]
> I think I have the labels right for this, but please let me know otherwise

This uses the "minor" label, which will increment the minor version in the draft release to 2.2. I don't think that is necessary for this change. This could be omitted or changed to "patch".

Also, we should merge this change into "master" (which should probably be renamed to "2") as well. Then we could start a "2.0" branch off of master ("2"), to release this patch for sites that still use compass syntax. Then we could merge the dart syntax (2.1 changes) into master ("2"), and rebase the "3" branch on top of it, so the dart sass changes carry over into the next major release.

Let me know if that makes sense, as I could see how a "2" branch could be confused with a "2.0" or "2.0.x" branch.

**Comment by aletail** [comment]
> This uses the "minor" label, which will increment the minor version in the draft release to 2.2. I don't think that is necessary for this change. This could be omitted or changed to "patch".

I like the idea of always using a label, I will change it to patch as that makes sense to me

> Also, we should merge this change into "master" (which should probably be renamed to "2") as well. Then we could start a "2.0" branch off of master ("2"), to release this patch for sites that still use compass syntax. Then we could merge the dart syntax (2.1 changes) into master ("2"), and rebase the "3" branch on top of it, so the dart sass changes carry over into the next major release.

So this makes sense, we should look into some sort of automations here. So maybe at the very least, once the original PR is approved and merged into it's intended branch, we auto create PRs to be merged into the correct branches. (Something like this would make things a bit easier/automatic, but not sure the best approach though.

Also, shouldn't  we have set the dart stuff to 3.0 instead of 2.1? I feel like that is a major change


Moving forward, I think we should site down and determine all the branches for this haha

**Comment by tiller1010** [comment]
@aletail
This all looks good!

> So this makes sense, we should look into some sort of automations here. So maybe at the very least, once the original PR is approved and merged into it's intended branch, we auto create PRs to be merged into the correct branches. (Something like this would make things a bit easier/automatic, but not sure the best approach though.

That's a good idea. Maybe when a PR is merged with a "patch" label, we create PRs for both the major branch (in this case, the master (2) branch) and every previous minor branch within the same major version (in this case, the hypothetical "2.0.x" branch). If the PR has a "minor" label, we create a single PR into the major branch (in this case, the 2.1 dart sass changes are set to merge into master (2), but not the "2.0.x" branch).

> Also, shouldn't we have set the dart stuff to 3.0 instead of 2.1? I feel like that is a major change

I didn't see this as much of a major change. We could have assumed that a majority of people might have used the webpack sass-loader, which works with dart sass syntax (and the deprecated compass syntax too, only with warnings). The sass also functions in the same way, and it is easy to switch out the compass compiler for the dart sass one on a site. That being said, I do believe the dart sass syntax does belong with the next major version, but releasing this minor version lets us use it now. The maintainers of this repo also considered this to be a "minor" feature https://github.com/jonsuh/hamburgers/compare/v1.1.3...v1.2.0

**Comment by aletail** [comment]
I think for simplicity - I am going to create a separate tasks to update the branches here. This will just be merged into 2.1 for now

**Comment by aletail** [comment]
@tiller1010 Do we need to manually bump the version number in package.json? If so, what do we bump it too?

**Comment by tiller1010** [comment]
@aletail 
> @tiller1010 Do we need to manually bump the version number in package.json? If so, what do we bump it too?

No, this should draft a "2.1.1" release and that will be inserted into the package.json https://github.com/werkbot/framewerk/blob/2.1/.github/workflows/draftrelease.yml#L38

**Pull request merged**
